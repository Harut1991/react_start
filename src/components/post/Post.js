// @flow

import React from 'react';

type Props = {|
    title: string,
    body: string,
|};

function Post(props: Props) {
    return (
        <li>
            <div className="card text-center">
                <div className="card-header">
                    {props.title}
                </div>
                <div className="card-body">
                    <h5 className="card-title">{props.body}</h5>
                </div>
            </div>
        </li>
    )

}
export default Post;
