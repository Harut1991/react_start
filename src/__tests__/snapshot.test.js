import renderer from "react-test-renderer";
import PhotoComponent from "../components/photo/photoComponent";
import React from "react";
import Post from "../components/post/Post";

test('Photo component test', () => {
    const component = renderer.create(
        <PhotoComponent thumbnailUrl="http://test" title="test" url="http://test" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Post component test', () => {
    const component = renderer.create(
        <Post body="test" title="test" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});
