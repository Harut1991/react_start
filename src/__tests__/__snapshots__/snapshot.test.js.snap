// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo component test 1`] = `
<div
  className="card"
>
  <img
    alt="Card image cap"
    className="card-img-top"
    src="http://test"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      test
    </h5>
    <a
      className="btn btn-primary"
      href="http://test"
    >
      Go to site
    </a>
  </div>
</div>
`;

exports[`Post component test 1`] = `
Array [
  <div
    className="card-header"
  >
    test
  </div>,
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      test
    </h5>
  </div>,
]
`;
